Opening a Java file with IDEA (dbl-click on file) causes an exception

```
java.lang.RuntimeException: Cannot invoke (class=ConnectDocumentToLanguageServerSetupParticipant, method=fileOpened, topic=FileEditorManagerListener)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:676)
	at com.intellij.util.messages.impl.MessageBusImplKt.deliverMessage(MessageBusImpl.kt:422)
	at com.intellij.util.messages.impl.MessageBusImplKt.pumpWaiting(MessageBusImpl.kt:401)
	at com.intellij.util.messages.impl.MessageBusImplKt.access$pumpWaiting(MessageBusImpl.kt:1)
	at com.intellij.util.messages.impl.MessagePublisher.invoke(MessageBusImpl.kt:460)
	at jdk.proxy1/jdk.proxy1.$Proxy88.fileOpened(Unknown Source)
	at com.intellij.ide.lightEdit.LightEditTabs.addEditorTab(LightEditTabs.java:105)
	at com.intellij.ide.lightEdit.LightEditTabs.addEditorTab(LightEditTabs.java:85)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.addEditorTab(LightEditServiceImpl.java:249)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.doOpenFile(LightEditServiceImpl.java:183)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.lambda$restoreSession$6(LightEditServiceImpl.java:510)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.lambda$restoreSession$7(LightEditServiceImpl.java:506)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.invokeOnEdt(LightEditServiceImpl.java:170)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.doWhenActionManagerInitialized(LightEditServiceImpl.java:164)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.restoreSession(LightEditServiceImpl.java:504)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.lambda$init$0(LightEditServiceImpl.java:104)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.invokeOnEdt(LightEditServiceImpl.java:170)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.init(LightEditServiceImpl.java:97)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.doShowEditorWindow(LightEditServiceImpl.java:128)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.doOpenFile(LightEditServiceImpl.java:178)
	at com.intellij.ide.lightEdit.LightEditServiceImpl.lambda$openFile$1(LightEditServiceImpl.java:150)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:209)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:191)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:862)
	at com.intellij.openapi.application.impl.ApplicationImpl$4.run(ApplicationImpl.java:482)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithEnabledImplicitRead(RwLockHolder.kt:75)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithImplicitRead(RwLockHolder.kt:67)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWithImplicitRead(ApplicationImpl.java:1449)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:82)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:124)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:44)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:318)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:792)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:739)
	at java.desktop/java.awt.EventQueue$3.run(EventQueue.java:733)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:761)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.kt:695)
	at com.intellij.ide.IdeEventQueue._dispatchEvent$lambda$12(IdeEventQueue.kt:589)
	at com.intellij.openapi.application.impl.RwLockHolder.runWithoutImplicitRead(RwLockHolder.kt:44)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.kt:589)
	at com.intellij.ide.IdeEventQueue.access$_dispatchEvent(IdeEventQueue.kt:72)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:355)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1$1.compute(IdeEventQueue.kt:354)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:793)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:354)
	at com.intellij.ide.IdeEventQueue$dispatchEvent$processEventRunnable$1$1.invoke(IdeEventQueue.kt:349)
	at com.intellij.ide.IdeEventQueueKt.performActivity$lambda$1(IdeEventQueue.kt:1014)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueueKt.performActivity(IdeEventQueue.kt:1014)
	at com.intellij.ide.IdeEventQueue.dispatchEvent$lambda$7(IdeEventQueue.kt:349)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:867)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.kt:391)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
Caused by: java.lang.IllegalStateException: InSmartMode can't be used in LightEdit mode, check that LightEdit.owns(project)==false before calling
	at com.intellij.openapi.application.impl.InSmartMode.<init>(AppUIExecutorImpl.kt:172)
	at com.intellij.openapi.application.impl.NonBlockingReadActionImpl.inSmartMode(NonBlockingReadActionImpl.java:146)
	at com.redhat.devtools.intellij.lsp4ij.internal.PromiseToCompletableFuture.nonBlockingReadActionPromise(PromiseToCompletableFuture.java:150)
	at com.redhat.devtools.intellij.lsp4ij.internal.PromiseToCompletableFuture.init(PromiseToCompletableFuture.java:79)
	at com.redhat.devtools.intellij.lsp4ij.ConnectDocumentToLanguageServerSetupParticipant$ConnectToLanguageServerCompletableFuture.<init>(ConnectDocumentToLanguageServerSetupParticipant.java:41)
	at com.redhat.devtools.intellij.lsp4ij.ConnectDocumentToLanguageServerSetupParticipant.fileOpened(ConnectDocumentToLanguageServerSetupParticipant.java:69)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeMethod(MessageBusImpl.kt:699)
	at com.intellij.util.messages.impl.MessageBusImplKt.invokeListener(MessageBusImpl.kt:663)
	... 61 more
```

Useful hint:
> InSmartMode can't be used in LightEdit mode, check that LightEdit.owns(project)==false before calling

This happened with 2023.3 EAP (Build #IC-233.9802.14, built on October 12, 2023)

--- a/src/main/java/com/redhat/devtools/intellij/lsp4ij/ContentTypeToLanguageServerDefinition.java
+++ b/src/main/java/com/redhat/devtools/intellij/lsp4ij/ContentTypeToLanguageServerDefinition.java
@@ -20,7 +20,7 @@ public class ContentTypeToLanguageServerDefinition extends AbstractMap.SimpleEnt
     }
 
     public boolean match(VirtualFile file, Project project) {
-        return documentMatcher.match(file, project);
+        return getValue().supportsCurrentEditMode(project) && documentMatcher.match(file, project);
     }
 
     public boolean shouldBeMatchedAsynchronously(Project project) {
@@ -32,6 +32,9 @@ public class ContentTypeToLanguageServerDefinition extends AbstractMap.SimpleEnt
     }
 
     public @NotNull <R> CompletableFuture<Boolean> matchAsync(VirtualFile file, Project project) {
+        if (!getValue().supportsCurrentEditMode(project)) {
+            return CompletableFuture.completedFuture(false);
+        }
         return documentMatcher.matchAsync(file, project);
     }
 }

--- a/src/main/java/com/redhat/devtools/intellij/lsp4ij/LanguageServersRegistry.java
+++ b/src/main/java/com/redhat/devtools/intellij/lsp4ij/LanguageServersRegistry.java
@@ -11,10 +11,9 @@
 package com.redhat.devtools.intellij.lsp4ij;
 
 import com.intellij.icons.AllIcons;
+import com.intellij.ide.lightEdit.LightEdit;
 import com.intellij.lang.Language;
-import com.intellij.openapi.editor.Document;
 import com.intellij.openapi.project.Project;
-import com.intellij.openapi.vfs.VirtualFile;
 import com.redhat.devtools.intellij.lsp4ij.client.LanguageClientImpl;
 import com.redhat.devtools.intellij.lsp4ij.server.StreamConnectionProvider;
 import org.eclipse.lsp4j.jsonrpc.Launcher;
@@ -55,9 +54,11 @@ public class LanguageServersRegistry {
         public final int lastDocumentDisconnectedTimeout;
         private boolean enabled;
 
+        public final boolean supportsLightEdit;
+
         final @Nonnull Scope scope;
 
-        public LanguageServerDefinition(@Nonnull String id, @Nonnull String label, String description, boolean isSingleton, Integer lastDocumentDisconnectedTimeout, String scope) {
+        public LanguageServerDefinition(@Nonnull String id, @Nonnull String label, String description, boolean isSingleton, Integer lastDocumentDisconnectedTimeout, String scope, boolean supportsLightEdit) {
             this.id = id;
             this.label = label;
             this.description = description;
@@ -65,6 +66,7 @@ public class LanguageServersRegistry {
             this.lastDocumentDisconnectedTimeout = lastDocumentDisconnectedTimeout != null && lastDocumentDisconnectedTimeout > 0 ? lastDocumentDisconnectedTimeout : DEFAULT_LAST_DOCUMENTED_DISCONNECTED_TIMEOUT;
             this.languageIdMappings = new ConcurrentHashMap<>();
             this.scope = scope == null || scope.isBlank()? Scope.application : Scope.valueOf(scope);
+            this.supportsLightEdit = supportsLightEdit;
             setEnabled(true);
         }
 
@@ -109,13 +111,17 @@ public class LanguageServersRegistry {
         public <S extends LanguageServer> Launcher.Builder<S> createLauncherBuilder() {
             return new Launcher.Builder<>();
         }
+
+        public boolean supportsCurrentEditMode(@NotNull Project project) {
+            return project != null && (supportsLightEdit || !LightEdit.owns(project));
+        }
     }
 
     static class ExtensionLanguageServerDefinition extends LanguageServerDefinition {
         private final ServerExtensionPointBean extension;
 
         public ExtensionLanguageServerDefinition(ServerExtensionPointBean element) {
-            super(element.id, element.label, element.description, element.singleton, element.lastDocumentDisconnectedTimeout, element.scope);
+            super(element.id, element.label, element.description, element.singleton, element.lastDocumentDisconnectedTimeout, element.scope, element.supportsLightEdit);
             this.extension = element;
         }
 

--- a/src/main/java/com/redhat/devtools/intellij/lsp4ij/ServerExtensionPointBean.java
+++ b/src/main/java/com/redhat/devtools/intellij/lsp4ij/ServerExtensionPointBean.java
@@ -42,6 +42,9 @@ public class ServerExtensionPointBean extends BaseKeyedLazyInstance<StreamConnec
     @Attribute("singleton")
     public boolean singleton;
 
+    @Attribute("supportsLightEdit")
+    public boolean supportsLightEdit;
+
     @Attribute("lastDocumentDisconnectedTimeout")
     public Integer lastDocumentDisconnectedTimeout;
 

--- a/src/main/java/com/redhat/devtools/intellij/lsp4ij/internal/PromiseToCompletableFuture.java
+++ b/src/main/java/com/redhat/devtools/intellij/lsp4ij/internal/PromiseToCompletableFuture.java
@@ -10,6 +10,7 @@
  ******************************************************************************/
 package com.redhat.devtools.intellij.lsp4ij.internal;
 
+import com.intellij.ide.lightEdit.LightEdit;
 import com.intellij.openapi.Disposable;
 import com.intellij.openapi.application.ReadAction;
 import com.intellij.openapi.progress.EmptyProgressIndicator;
@@ -74,8 +75,8 @@ public class PromiseToCompletableFuture<R> extends CompletableFuture<R> {
     }
 
     protected void init() {
-        // if indexation is processing, we need to execute the promise in smart mode
-        var executeInSmartMode = DumbService.getInstance(project).isDumb();
+        // if indexation is processing and not in Light Edit mode, we need to execute the promise in smart mode
+        var executeInSmartMode =  !LightEdit.owns(project) && DumbService.getInstance(project).isDumb();
         var promise = nonBlockingReadActionPromise(executeInSmartMode);
         bind(promise);
     }

